tokio/src/sync/task/atomic_waker.rs

```diff
@@ -238,13 +255,27 @@ impl AtomicWaker {
                 debug_assert!(state == REGISTERING || state == REGISTERING | WAKING);
             }
         }
+
+        struct PanicGuard<'a>(&'a AtomicUsize);
+
+        impl Drop for PanicGuard<'_> {
+            fn drop(&mut self) {
+                // On panics, we restore straight back to WAITING. We just tried
+                // to replace the atomic waker, so it's a legitimate outcome
+                // that the net effect of the operation is a no-op (i.e. no
+                // wakeup was issued).
+                self.0.swap(WAITING, AcqRel);
+            }
+        }
```

Comment:
Why is this a swap and not just a store?

Reply:
No, it should just be `store`. I think this is a typo that came out of copying this:
https://github.com/tokio-rs/tokio/blob/master/tokio/src/sync/task/atomic_waker.rs#L210

Thanks!